### PR TITLE
{vis}[fosscuda/2018b] PyQt v4.12.3 w/ Python 2.7.15

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)
Depends on ~~#7645, #7644~~

### CODE REVIEW

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/82bff63487ee47d0769a3dca6cd27814 for a full test report.

**Comment by akesandgren** [comment]
There may be some problem with this one.
A simple import fails with

``python -c 'import PyQt4.QtHelp'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named sip
```
but
```
python -c 'import sip'
```
works

**Comment by akesandgren** [comment]
It is also installing things where it shouldn't...

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/cbd96c73b9e066c8710e09fb638d5fda for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/2c15a90372516d58460c8b2e73098056 for a full test report.

**Comment by boegel** [comment]
@akesandgren Please add the missing checksums (you'll need to do that manually, `--inject-checksums` doesn't work yet on `components`...), and fix the code style issue:

```
$ eb --from-pr 7646 --check-style
== temporary log file in case of crash /tmp/eb-KmWk5K/easybuild-73UatH.log

Running style check on 1 easyconfig(s)...

/tmp/eb-KmWk5K/files_pr7646/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb:58:15: E131 continuation line unaligned for hanging indent
[FAIL] /tmp/eb-KmWk5K/files_pr7646/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb

ERROR: One or more style checks FAILED!

File: easybuild/easyconfigs/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+pyqtcfgopts = 'configure-ng.py --confirm-license --verbose --no-python-dbus '
+pyqtcfgopts += common_cfgopts + ' --sip=%(installdir)s/bin/sip '
+pyqtcfgopts += '--designer-plugindir=%(installdir)s/plugins/designer '
+
+# PyQt needs a private copy of the sip module so we need to do this as a Bundle.
+components = [
+    ('SIP', '4.19.13', {
+        'source_urls': ['http://sourceforge.net/projects/pyqt/files/sip/sip-%(version)s'],
+        'sources': [SOURCELOWER_TAR_GZ],
+        'checksum': 'e353a7056599bf5fbd5d3ff9842a6ab2ea3cf4e0304a0f925ec5862907c0d15e',
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb]: @akesandgren Should be `checksums`:

```
ERROR: Failed to get application instance for PyQt (easyblock: Bundle): Use of unknown easyconfig parameter 'checksum' when setting parameter value to 'e353a7056599bf5fbd5d3ff9842a6ab2ea3cf4e0304a0f925ec5862907c0d15e'
```

File: easybuild/easyconfigs/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        'start_dir': 'sip-%(version)s',
+        'configopts': sipcfgopts,
+        'options': {'modulename': 'PyQt4.sip'},
+    }),
+    (name, version, {
+        'source_urls': ['http://sourceforge.net/projects/pyqt/files/%(name)s%(version_major)s/%(name)s-%(version)s'],
+        'sources': ['%(name)s%(version_major)s_gpl_x11-%(version)s.tar.gz'],
+        'checksum': 'a00f5abef240a7b5852b7924fa5fdf5174569525dc076cd368a566619e56d472',
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb]: same here, should be `checksums`

File: easybuild/easyconfigs/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+pyqtcfgopts = 'configure-ng.py --confirm-license --verbose --no-python-dbus '
+pyqtcfgopts += common_cfgopts + ' --sip=%(installdir)s/bin/sip '
+pyqtcfgopts += '--designer-plugindir=%(installdir)s/plugins/designer '
+
+# PyQt needs a private copy of the sip module so we need to do this as a Bundle.
+components = [
+    ('SIP', '4.19.13', {
+        'source_urls': ['http://sourceforge.net/projects/pyqt/files/sip/sip-%(version)s'],
+        'sources': [SOURCELOWER_TAR_GZ],
+        'checksums': 'e353a7056599bf5fbd5d3ff9842a6ab2ea3cf4e0304a0f925ec5862907c0d15e',
```
Comment (boegel) [inline easybuild/easyconfigs/p/PyQt/PyQt-4.12.3-fosscuda-2018b-Python-2.7.15.eb]: @akesandgren This should be a list (same below):

```python
    ('SIP', '4.19.13', {
        ...
        'checksums': ['e353a7056599bf5fbd5d3ff9842a6ab2ea3cf4e0304a0f925ec5862907c0d15e'],
```

(same below)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2402.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/37eab0d376c07677c50b1a47b8288616 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2119.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/cff8e1c39abd6048f301a3ea7699cfca for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3108.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/fee01cd397ee75566855a3906dd8a60f for a full test report.

**Comment by boegel** [comment]
Going in, thanks @akesandgren!

**Pull request merged**